### PR TITLE
add support for CREATE TEMPORARY TABLE

### PR DESCRIPTION
**Pull request by leiysky**
### What problem does this PR solve? 
Add support for CREATE TEMPORARY TABLE #408 

### What is changed and how it works?
MySQL syntax:
```
create_table_stmt:
          CREATE opt_temporary TABLE_SYM opt_if_not_exists table_ident
          '(' table_element_list ')' opt_create_table_options_etc
        | CREATE opt_temporary TABLE_SYM opt_if_not_exists table_ident
          opt_create_table_options_etc
```

### Check List 

Tests 

 - Unit test

### CODE REVIEW

**Comment by tangenta** [comment]
@leiysky TiDB does not support `TEMPORARY` tables. Would you please add a warning in the parser for both `CreateTableStmt` and `DropTableStmt`?

**Comment by leiysky** [comment]
@tangenta PTAL

**Comment by tangenta** [comment]
@leoppro PTAL

File: parser.y [hunk 1]
```diff
@@ -2457,25 +2457,35 @@ DatabaseOptionList:
...
-		stmt.Table = $4.(*ast.TableName)
-		stmt.IfNotExists = $3.(bool)
-		stmt.Options = $6.([]*ast.TableOption)
-		if $7 != nil {
-			stmt.Partition = $7.(*ast.PartitionOptions)
+		stmt := $6.(*ast.CreateTableStmt)
+		stmt.Table = $5.(*ast.TableName)
+		stmt.IfNotExists = $4.(bool)
+		stmt.IsTemporary = $2.(bool)
+		if $2.(bool) {
```
Comment (leoppro) [inline parser.y]:
```suggestion
		if stmt.IsTemporary {
```

Comment (leoppro) [inline parser.y]:
```suggestion
		if stmt.IsTemporary {
```

File: parser.y [hunk 2]
```diff
@@ -2457,25 +2457,35 @@ DatabaseOptionList:
...
+			yylex.AppendError(yylex.Errorf("TiDB doesn't support CREATE TEMPORARY TABLE, TEMPORARY will be parsed but ignored."))
```
Comment (leoppro) [inline parser.y]: could you throw a warn at line 3081?

File: parser.y [hunk 3]
```diff
@@ -2457,25 +2457,35 @@ DatabaseOptionList:
...
+			parser.lastErrorAsWarn()
+		}
+		stmt.Options = $7.([]*ast.TableOption)
+		if $8 != nil {
```
Comment (leoppro) [inline parser.y]: why we need to check `$8`?

File: parser_test.go [hunk 4]
```diff
@@ -1766,6 +1766,8 @@ func (s *testParserSuite) TestDDL(c *C) {
 		{"CREATE TABLE foo (a.b, b);", false, ""},
 		{"CREATE TABLE foo (a, b.c);", false, ""},
 		{"CREATE TABLE (name CHAR(50) BINARY)", false, ""},
+		// for create temporary table
+		{"CREATE TEMPORARY TABLE t (a varchar(50), b int);", true, "CREATE TEMPORARY TABLE `t` (`a` VARCHAR(50),`b` INT)"},
```
Comment (leoppro) [inline parser_test.go]: could you add more test cases, cover most of the branches

File: parser.y (hunk 3 above)
Comment (leiysky) [reply parser.y]: It would be `nil interface{}` if there is no `PartitionOpt`, and the type assertion would panic.

File: parser_test.go (hunk 4 above)
Comment (leiysky) [reply parser_test.go]: added

**Pull request merged**